frontend/javascripts/oxalis/api/api_latest.ts

```diff
@@ -702,12 +704,15 @@ class TracingApi {
       existingMagIndex,
     );
     const [dx, dy, dz] = currentMag;
-
+    // getDataForBoundingBox grows the bounding box to be mag-aligned which can change the dimensions
+    const boundingBoxInMag1MagAligned = boundingBoxInMag1.alignWithMag(currentMag, "grow");
```

Comment:
Just for my understanding, where can I find the code in which `getDataForBoundingBox` might be growing the bbox?

Reply:
It's quite hidden because the code is complex, which is unfortunate. `getDataForBoundingBox` requests all buckets that intersect with the mag 1 bounding box (`getBucketAddressesInCuboid`) so the gathered data usually is larger than the mag 1 bounding box, because the bounding box is not aligned to 32³ buckets. Then, `cutOutCuboid` is called which converts the mag 1 bounding box to the request mag by using the `scaleGlobalPositionWithMagnification` method. This method has a `ceil` argument which is not set for top left (so that is floored) and set for bottom right. This possibly grows the bounding box (when thinking about it in mag 1 coordinates, which is what is done to determine the segment positions).
Does that make sense?

Reply:
Okay, that makes sense! I understand that both in `cutOutCuboid`->`scaleGlobalPositionWithMagnification` the topLeft is floored and the bottomRight is ceiled, which is the same as what is happening in `alignWithMag` when it's called with the argument `"grow"`.